src/http/request_template.rs

```diff
@@ -114,12 +114,9 @@ impl RequestTemplate {
           req.body_mut().replace(body_path.render(ctx).into());
         }
         Encoding::ApplicationXWwwFormUrlencoded => {
-          // TODO: this is a performance bottleneck
-          // We first encode everything to string and then back to form-urlencoded
-          let body: String = body_path.render(ctx);
-          let form_data = match serde_json::from_str::<serde_json::Value>(&body) {
+          let form_data = match serde_json::from_str::<serde_json::Value>(&body_path.render(ctx)) {
             Ok(deserialized_data) => serde_urlencoded::to_string(deserialized_data)?,
-            Err(_) => body,
+            Err(_) => body_path.render(ctx),
```

Comment:
This is basically the same code, no?

Reply:
since above you have `body_path.render(ctx)` there is no optimization and you convert to string anyway. 

To solve the issue you need to remove `from_str` calls completely. Check that Mustache have different methods for rendering like `render_graphql` to support different string output

Reply:
Okay, got it